src/odemis/driver/lakeshore.py

```diff
@@ -344,98 +411,220 @@ def HeaterSetup(self,
-            self.checkError()
+            for va_name, sensor_name in self._sensor_inputs.items():
+                temp = self.GetSensorTemperature(sensor_name)
+                if va_name == "":
+                    full_va_name = "temperature"
+                else:
+                    full_va_name = f"{va_name}Temperature"
+                va = getattr(self, full_va_name)
+                va._set_value(temp, force_write=True)
+                logging.debug(u"Lakeshore temperature: %f °C", va.value)
+                self.checkError()
         except:
             # another exception.
             logging.exception("Failed to read sensor temperature.")
 
 
+class SerialBusAccesser(object):
+    """
+    Manages connections to the low-level bus
+    """
+
+    def __init__(self, port: str):
+        # Ensure no one will talk to it simultaneously, and we don't talk to devices already in use
+        """
+        port: the full path to the serial port (eg: /dev/ttyUSB1)
+        raises:
+           IOError: if something went wrong opening the port
+        """
+        self._ser_access = threading.Lock()
+
+        # For debugging purpose
+        if port == "/dev/fake":
+            self._serial = LakeshoreSimulator(timeout=1)
+            self._file = None
+
+            return
+
+        self._file = open(port)  # Open in RO, just to check for lock
+        fcntl.flock(self._file.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)  # Raises IOError if cannot lock
+
+        self._serial = self._openSerialPort(port)
+
+    def terminate(self):
+        with self._ser_access:
+            self._serial.close()
+
+    def sendOrder(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF)
+        """
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+            time.sleep(0.05)
+
+    def sendQuery(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF, but with the ?)
+        returns (byte str): answer received from the device (without \n or \r)
+        raise:
+            IOError if no answer is returned in time
+        """
+
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+
+            self._serial.timeout = 1
+            ans = b''
+            while ans[-1:] != b'\n':
+                char = self._serial.read()
+                if not char:
+                    raise IOError("Timeout after receiving %s" % to_str_escape(ans))
+                ans += char
+
+            logging.debug("Received answer %s", to_str_escape(ans))
+
+            return ans.strip()
+
+
+class IPBusAccesser(object):
+    """
+    Manages connections to the low-level bus
+    """
+
+    def __init__(self, ip_address):
```

Comment:
Please add a docstring to specify the argument. BTW, for new code, you can use the typing annotations. Like `__init__(self, ip_address: str):`